File: 2487-optimal-partition-of-string/optimal-partition-of-string.java
class Solution {
    public int partitionString(String s) {
        List<String> al = new ArrayList<>();
        String temp = "";
        for (int i=0; i < s.length(); i++) {
            if (!temp.contains(String.valueOf(s.charAt(i))))
                temp = temp + String.valueOf(s.charAt(i));
            else {
                al.add(temp);
                temp = String.valueOf(s.charAt(i));
            }
        }
        al.add(temp);

        return al.size();
    }
}

File: 2487-optimal-partition-of-string/optimal-partition-of-string.py
class Solution:
    def partitionString(self, s: str) -> int:
        temp, result = '', []

        for i in s:
            if i not in temp:
                temp += i
            else:
                result.append(temp)
                temp = i
        else:
            result.append(temp)

        print(result)
        return len(result)